apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SessionPool.TargetedSessionPool.cs

```diff
@@ -483,8 +483,8 @@ void RemoveStaleOrExpiredItemsFromQueue(ConcurrentQueue<PooledSession> queue)
                         }
                         else
                         {
-                            // The session is fine - put it back on the queue.
-                            queue.Enqueue(session);
+                            // The session is fine - put it back on the stack.
+                            stack.Push(session);
```

Comment:
We can't do this - we'll end up pushing and popping the same element multiple times.
We could do with a dequeue really - but otherwise, we'll need to work out how to check everything but end up with it in the right order.

Before fixing this, it would be good to have a test for this case (that fails with the current code) if possible.

Reply:
Ohh, rigth, totally went over this. Will right test first and fix after. Thanks for catching!

Reply:
This is now fixed.
Before fixing, the new test failed as expected.